src/app/components/ImageTable.tsx

```diff
@@ -35,15 +37,135 @@ export const ImageTable: React.FunctionComponent = () => {
   const [perPage, setPerPage] = React.useState(20);
   const [isDrawerExpanded, setIsExpanded] = React.useState(false);
   const [selectedImage, setSelectedImage] = React.useState({});
+
+  const [providerSelections, setProviderSelections] = React.useState<string[]>([]);
+  const [regionSelections, setRegionSelections] = React.useState<string[]>([]);
+  const [architectureSelections, setArchitectureSelections] = React.useState<string[]>([]);
+  const [filterConfig, setFilterConfig] = React.useState<object[]>([]);
+
   const drawerRef = React.useRef<HTMLDivElement>();
 
+  const onFilter = (image: ImageData) => {
+    const matchesProviderValue = providerSelections.includes(image.provider.toLowerCase());
+    const matchesRegionValue = regionSelections.includes(image.region.toLowerCase());
+    const matchesArchitectureValue = architectureSelections.includes(image.arch.toLowerCase());
+
+    return (
+      (providerSelections.length === 0 || matchesProviderValue)
+      && (regionSelections.length === 0 || matchesRegionValue)
+      && (architectureSelections.length === 0 || matchesArchitectureValue)
+    );
+  };
+
+  const onFilterSelect = (_: React.MouseEvent | undefined, itemId: string | number | undefined) => {
```

Comment:
General TS question: Jest won't let my tests pass if I won't specify the entire array of possible inputs for functions that are derived from "onSelect" which in this case is "string | number | undefined". Is there a better way to declare this on a function that only accepts strings anyways?

Reply:
which test are you referring to?

Reply:
I tried a few things, couldn't make it work without respecting the mentioned inputs

Reply:
Thanks for giving it a try!

Reply:
The only shorter version I could come up with would be using a lookup type (or indexed access type) on `MenuProps['onSelect']`. This would still mean that the `itemId` can be `string | number | undefined`, but you wouldn't have to write the types all over again. This can also already happen on [the ImageTableFilter onFilterSelect prop](https://github.com/redhatcloudx/cloud-image-directory-frontend/blob/175dc3929436d80827a0f066fca199c82ef5ea7d/src/app/components/ImageTableFilter.tsx#L15).

```suggestion
  // import { MenuProps } from '@patternfly/react-core'
  // ...
  const onFilterSelect: MenuProps['onSelect'] = (_, itemId) => {
```